1-mandatory/3-slideshow/slideshow.js

```diff
@@ -1 +1,20 @@
 // Write your code here
+const buttons = document.querySelectorAll("[data-carousel-button]")
+
+buttons.forEach(button => {
+  button.addEventListener("click", () => {
+    const offset = button.dataset.carouselButton === "next" ? 1 : -1
+    const slides = button
+      .closest("[data-carousel]")
+      .querySelector("[data-slides]")
+
+    const activeSlide = slides.querySelector("[data-active]")
+    let newIndex = [...slides.children].indexOf(activeSlide) + offset
+    if (newIndex < 0) newIndex = slides.children.length - 1
```

Comment:
Good work! This code works which is brillant, but I would add some comments: it's fairly complex, I would also format the ifs using {} for readablity